### PR TITLE
Update style and rule enforcement in .editorconfig

### PR DESCRIPTION
**Pull request by davkean**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -140,6 +138,18 @@ csharp_style_pattern_matching_over_as_with_null_check = true:suggestion
...
+csharp_style_deconstructed_variable_declaration = true:suggestion
+dotnet_style_object_initializer = true:suggestion
+dotnet_style_collection_initializer = true:suggestion
+dotnet_style_coalesce_expression = true:suggestion
+dotnet_style_null_propagation = true:suggestion
+dotnet_style_explicit_tuple_names = true:suggestion
+dotnet_style_prefer_inferred_tuple_names = true:suggestion
+dotnet_style_prefer_inferred_anonymous_type_member_names = true:suggestion
+dotnet_style_prefer_is_null_check_over_reference_equality_method = true:suggestion
+dotnet_style_prefer_conditional_expression_over_return = false
```
Comment (davkean) [inline .editorconfig]: This one doesn't work: https://github.com/dotnet/roslyn/issues/27197

Comment (Pilchie) [reply .editorconfig]: Try `false:none`?  Not sure how things work if you don't have a severity.

Comment (davkean) [reply .editorconfig]: Reading the code, `false` is allowed. I tried `false:none` it results in the same behavior. I spent ~30 minutes trying to figure this out and gave up. My small repro stopped occurring once I attached a debugger.

**Comment by davkean** [comment]
I'm going to merge this in without figuring out `dotnet_style_prefer_conditional_expression_over_return` - I'll make a followup change once we figure it out.

**Pull request merged**